tests/monotouch-test/ARKit/ARPlaneGeometryTest.cs

```diff
@@ -0,0 +1,145 @@
+﻿//
+// Unit tests for ARPlaneGeometry
+//
+// Authors:
+//	Vincent Dondain <vidondai@microsoft.com>
+//
+// Copyright 2018 Microsoft. All rights reserved.
+//
+
+#if XAMCORE_2_0 && __IOS__
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using ARKit;
+using Foundation;
+using NUnit.Framework;
+using ObjCRuntime;
+
+using VectorFloat2 = global::OpenTK.Vector2;
+using VectorFloat3 = global::OpenTK.NVector3;
+
+namespace MonoTouchFixtures.ARKit {
+
+	class ARPlaneGeometryPoker : ARPlaneGeometry {
+
+		GCHandle verticesArrayHandle;
+		GCHandle textureCoordinatesArrayHandle;
+		GCHandle indicesArrayHandle;
+		GCHandle boundaryVerticesArrayHandle;
+		VectorFloat3 [] vertices;
+		VectorFloat2 [] textureCoordinates;
+		short [] indices;
+		VectorFloat3 [] boundaryVertices;
+
+		public ARPlaneGeometryPoker () : base (IntPtr.Zero)
+		{
+		}
+
+		public override nuint VertexCount => 2;
+
+		public override nuint TextureCoordinateCount => 2;
+
+		// There are always 3x more 'TriangleIndices' than 'TriangleCount' since 'TriangleIndices' represents Triangles (set of three indices).
+		// So 2 'TriangleCount' = 6 'TriangleIndices'.
+		public override nuint TriangleCount => 2;
+
+		public override nuint BoundaryVertexCount => 2;
+
+		public override IntPtr GetRawVertices ()
+		{
+			vertices = new VectorFloat3 [] { new VectorFloat3 (1, 2, 3), new VectorFloat3 (4, 5, 6) };
+			if (!verticesArrayHandle.IsAllocated)
+				verticesArrayHandle = GCHandle.Alloc (vertices, GCHandleType.Pinned);
+			return verticesArrayHandle.AddrOfPinnedObject ();
+		}
+
+		public override IntPtr GetRawTextureCoordinates ()
+		{
+			textureCoordinates = new VectorFloat2 [] { new VectorFloat2 (1, 2), new VectorFloat2 (3, 4) };
+			if (!textureCoordinatesArrayHandle.IsAllocated)
+				textureCoordinatesArrayHandle = GCHandle.Alloc (textureCoordinates, GCHandleType.Pinned);
+			return textureCoordinatesArrayHandle.AddrOfPinnedObject ();
+		}
+
+		public override IntPtr GetRawTriangleIndices ()
+		{
+			// Two triangles (set of 3 indices)
+			indices = new short [] { 1, 2, 3, 4, 5, 6 };
```

Comment:
I'm a little confused, is this line creating an array of 6 shorts? can't you do new short [6]?? I might be missing something.

Reply:
Because I'm setting values that I can then test. See https://github.com/xamarin/xamarin-macios/pull/3343/files#diff-58636f8c30f5f14ba7b2793d5900e7cfR129

I'm simulating the fact that there are **actual** indices.